stock_restrict_lot/models/stock_move.py

```diff
@@ -88,3 +89,44 @@ def _split(self, qty, restrict_partner_id=False):
         if vals_list and self.restrict_lot_id:
             vals_list[0]["restrict_lot_id"] = self.restrict_lot_id.id
         return vals_list
+
+    def get_all_dest_moves(self):
+        res = self.move_dest_ids
+        moves_to_search = self.move_dest_ids
+        while moves_to_search:
+            new_dest_ids = moves_to_search.move_dest_ids
+            moves_to_search = new_dest_ids - res
+            res |= new_dest_ids
+        return res
+
+    def get_all_orig_moves(self):
+        res = self.move_orig_ids
+        moves_to_search = self.move_orig_ids
+        while moves_to_search:
+            new_orig_ids = moves_to_search.move_orig_ids
+            moves_to_search = new_orig_ids - res
+            res |= new_orig_ids
+        return res
+
+    def write(self, vals):
+        if "restrict_lot_id" not in vals:
+            return super().write(vals)
+        else:
+            restrict_lot_id = vals.pop("restrict_lot_id")
+            restrict_lot = self.env["stock.lot"].browse(restrict_lot_id)
+            chained_moves = self | self.get_all_dest_moves() | self.get_all_orig_moves()
+            if any(
+                [
+                    sm.state == "done" and sm.lot_ids != restrict_lot
```

Comment:
Not sure if I am the right person to review this anymore as I haven't worked with lots since $OLD version of Odoo, but logically, should it perhaps be allowed if the lot on the stock move is not set?
```suggestion
                    sm.state == "done" and sm.lot_ids and sm.lot_ids != restrict_lot
```

Reply:
done

Reply:
Sorry to intervene this late, but I think this check is not really useful...

I don't think it's possible to have a move done for a tracked product with lot_ids not set.
This should have raised a UserError telling "You need to supply a Lot/Serial Number for product..." (see https://github.com/odoo/odoo/blob/b2411f0109374958c317ed54c3156d8198130526/addons/stock/models/stock_move_line.py#L590)

Right ?

Reply:
I think you are perfectly right under normal circumstances. To me, it just looked funny to say 'the lot is not the same' without checking it is actually set. Anyway, now that it's done, does it hurt to check?

Reply:
Indeed it doesn't hurt !